ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -3096,6 +3096,8 @@ Seems much cleaner if each stmt is identified as a particular HiveOperation (whi
         assert t != null;
         if (AcidUtils.isTransactionalTable(t) && sharedWrite) {
           compBuilder.setSharedWrite();
+        } else if (MetaStoreUtils.isNonNativeTable(t.getTTable())) {
+          compBuilder.setLock(getLockTypeFromStorageHandler(output, t));
```

Comment:
do we need locks for iceberg delete/update. That should be handled in iceberg itself, isn't it?

Reply:
👍